lib/improver/utilities/cube_manipulation.py

```diff
@@ -193,15 +188,375 @@ def concatenate_cubes(
+    Warns:
+        Warning: If it does not know what to do with an unmatching
+                 attribute. Default is to delete it.
+    """
+    unmatching_attributes = compare_attributes(cubes)
+    if len(unmatching_attributes) > 0:
+        for i, cube in enumerate(cubes):
+            # Remove history.
+            if "history" in unmatching_attributes[i]:
+                cube.attributes.pop("history")
+                unmatching_attributes[i].pop("history")
+            # Normalise grid_id to ukx_standard_1
+            if "grid_id" in unmatching_attributes[i]:
+                if cube.attributes['grid_id'] in ['enukx_standard_v1',
+                                                  'ukvx_standard_v1']:
+                    cube.attributes['grid_id'] = 'ukx_standard_v1'
+                    unmatching_attributes[i].pop("grid_id")
+            # Add model_id if titles do not match.
+            if "title" in unmatching_attributes[i]:
+                model_title = cube.attributes.pop('title')
+                new_model_id_coord = build_coordinate([100*i],
+                                                      long_name='model_id',
+                                                      data_type=np.int)
+                new_model_coord = build_coordinate([model_title],
+                                                   long_name='model',
+                                                   coord_type=AuxCoord,
+                                                   data_type=np.str)
+                cube.add_aux_coord(new_model_id_coord)
+                cube.add_aux_coord(new_model_coord)
+                unmatching_attributes[i].pop("title")
+            # Remove any other mismatching attributes but raise warning.
+            if len(unmatching_attributes[i]) != 0:
+                for key in unmatching_attributes[i]:
+                    msg = ('Do not know what to do with ' + key +
+                           ' will delete it')
+                    warnings.warn(msg)
+                    cube.attributes.pop(key)
+
+
+def equalise_cube_coords(cubes):
+    """
+    Function to equalise coordinates that do not match.
+
+    Args:
+        cubes : Iris cubelist
+            List of cubes to check the coords and revise.
+    Returns:
+        cubelist : Iris cubelist
+            List of cubes with revised coords.
+    Raises:
+        ValueError: If Percentile coordinates do not match.
+        ValueError: If Threshold coordinates do not match.
+        ValueError: If model_id has more than one point.
+    """
+    unmatching_coords = compare_coords(cubes)
+    if len(unmatching_coords) == 0:
+        cubelist = cubes
+    else:
+        cubelist = iris.cube.CubeList([])
+        for i, cube in enumerate(cubes):
+            slice_over_keys = []
+            for key in unmatching_coords[i]:
+                # mismatching percentiles - raise error.
+                if key.find('percentile_over') > 0:
```

Comment:
Better to do `if 'percentile_over' in key` rather than find, unless I'm missing something - did you mean `>= 0` or not?